enhancements/ptp-ha.md

```diff
@@ -0,0 +1,250 @@
+---
+title: ptp_multi-nic_ha_support
+authors:
+  - "@danielmellado"
+reviewers:
+  - "@zshi-redhat"
+  - "@SchSeba"
+  - "@aneeshkp"
+  - "@imiller0"
+approvers:
+  - "@ijolliffe"
+creation-date: 2022-06-15
+last-updated: 2022-06-15
+tracking-link: TBD
+status: provisional
+---
+
+# PTP multi-nic HA Support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Multi-nic support for PTP operator [1] seems to be already built-in. In
+enterprise telco environments, there's often a need to load balance two
+different signals from different PTP IEEE 1588-2008 sources.
+
+PTP uses best master clock (BMC) algorithm to perform a selection of the
+candidate clock based on a series of parameters such as accuracy, quality,
+stratum and uuid. The HA support should also consider these ones and perform
+the change when needed.
+
+In Telco environments, where an specific profile, such as G.8275.1[] is needed,
+ABMC (Alternate BMC is required).
+
+## Scenarios
+
+### Multiple NICs
+
+This case assumes that over the that the Virtual Distributed Unit (vDU) hosts
+several NICs supporting Boundary Clock Function. The synchronization source is
+selected by *ptp4l*
+
+On Openshift 4.10 the PTP operator supports running multiple NICs with the
+following dual nic with two ptp4l configurations, but no HA on failure.
+
+```text
+        ┌─────────────────────────┐
+        │                         │
+        │                         │
+        │    Grand Master         │
+        │                         │
+        │                         │
+        └───────┬───────────┬─────┘
+                │           │
+                │           │
+                │           │
+         ┌──────┼───────────┼────┐
+         │ vDU  │           │    │
+         │ ┌────┴───┐  ┌────┴──┐ │
+         │ │        │  │       │ │
+         │ └─────┬──┘  └────┬──┘ │
+         │  NIC1 │      NIC2│    │
+         └───────┼──────────┼────┘
+                 │          │
+                 │          │
+          ┌──────┼──────────┼───┐
+          │      │          │   │
+          │      │       ┌──┘   │
+          │      │       │      │
+         ┌┼┬────┬┼┐     ┌┼┬────┬┼┐
+         ├─┘    └─┤     ├─┘    └─┤
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │        │     │        │
+         │ RU#0   │     │ RU#1   │
+         └────────┘     └────────┘
+```
+
```

Comment:
Does this mean that RU#0 will sync RU#1 in case NIC2 shut down (whether the problem is in NIC2, the NIC-peer in RU#1 or the link beween them)?

I belive that HA in this case should be NIC1 using two ports to connect to one NIC port of both RU#0 and RU#1 and NIC2 doing the same.

Another approach would be to have two NICs (with their PHC) in both RU, but still, connecting them as I mentioned before so that, their PHCs get synced via hardware. In such case, a handover must be done for phc2sys as it is selecting PHC in one of the NICs to set SYSTEM_CLOCK but now it must use the PHC in the other NIC.

The key point is that all synchronization must be via hardware. Otherwise, Max Time Error not greater than 30 ns won't be accomplished.